src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/ContentMapper.java

```diff
@@ -353,20 +357,20 @@ public ObjectMapper generateNewPreconfiguredContentMapper() {
         ChoiceDeserializer choiceDeserializer = new ChoiceDeserializer(contentDeserializer, itemDeserializer);
 
         QuestionValidationResponseDeserializer validationResponseDeserializer 
-            = new QuestionValidationResponseDeserializer(
-                contentDeserializer, choiceDeserializer);
+            = new QuestionValidationResponseDeserializer(contentDeserializer, choiceDeserializer);
 
-        IsaacQuestionBaseDeserializer isaacQuestionBaseDeserializer = new IsaacQuestionBaseDeserializer(contentDeserializer);
+        IsaacQuestionBaseDeserializer isaacQuestionBaseDeserializer =
+                new IsaacQuestionBaseDeserializer(contentDeserializer);
 
         SimpleModule contentDeserializerModule = new SimpleModule("ContentDeserializerModule");
         contentDeserializerModule.addDeserializer(ContentBase.class, contentDeserializer);
         contentDeserializerModule.addDeserializer(IsaacQuestionBase.class, isaacQuestionBaseDeserializer);
         contentDeserializerModule.addDeserializer(Choice.class, choiceDeserializer);
         contentDeserializerModule.addDeserializer(Item.class, itemDeserializer);
         contentDeserializerModule.addDeserializer(QuestionValidationResponse.class, validationResponseDeserializer);
+        contentDeserializerModule.addDeserializer(
+                LLMMarkingExpression.class, new LLMMarkingExpressionDeserializer(objectMapper));
 
```

Comment:
The ability to pass in a shared object mapper to all of these deserializers looks to be a pattern that could remove the need to order these deserializers in a special way to try to handle our recursive content model. We are only registering the deserializers for classes at this stage, the object mapper will handle the recursive content just fine when it comes to actually deserializing.